website/server/controllers/api-v3/groups.js

```diff
@@ -718,4 +715,19 @@ api.inviteToGroup = {
   },
 };
 
+// Transitioning invitation storage to an array of objects (see issue #7792 for reference)
+function clearPartyInvitation(user, groupID) {
+  user.invitations.parties.forEach(function(party, index, partiesArr) {
```

Comment:
Nice.  It's been a while since I used Mongoose but I just read if it's a mixed schema we need a markModified call [click for link](http://mongoosejs.com/docs/schematypes.html).  Let me know if I missed something!